Tasks/NpmAuthenticate/npmauth.ts

```diff
@@ -0,0 +1,136 @@
+import * as path from 'path';
+import * as tl from 'vsts-task-lib/task';
+import * as URL from 'url';
+import * as fs from 'fs';
+import * as constants_1 from './constants';
+import * as npmregistry_1 from 'npm-common/npmregistry'
+import * as util from 'npm-common/util';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+    let saveNpmrcPath;
+    let npmrc = tl.getInput(constants_1.NpmAuthenticateTaskInput.WorkingDir);
+    let workingDir = path.dirname(npmrc);
+    if (!(npmrc.endsWith('.npmrc') && tl.exist(npmrc))) {
+        throw new Error(tl.loc('NpmrcNotNpmrc', npmrc));
+    }
+    else {
+        console.log(tl.loc("AuthenticatingThisNpmrc", npmrc));
+    }
+
+    if (process.env.SAVE_NPMRC_PATH) {
+         saveNpmrcPath = process.env.SAVE_NPMRC_PATH
+    }
+    else {
+        let tempPath = tl.getVariable('Agent.BuildDirectory') || tl.getVariable('Agent.ReleaseDirectory') || process.cwd();
+        tempPath = path.join(tempPath, 'npmAuthenticate');
+        tl.mkdirP(tempPath);
+        saveNpmrcPath = fs.mkdtempSync(tempPath + path.sep); 
+        console.log("##vso[task.setvariable variable=SAVE_NPMRC_PATH;]" + saveNpmrcPath);
```

Comment:
Use `tl.setVariable()` instead of printing to console. Also, please update on the other places where we did this.